### PR TITLE
FI-385: Check for US Core profile support

### PR DESCRIPTION
**Pull request by Jammjammjamm**
Update the US Core Capability Statement sequence to include a test that checks for support of the US Core Patient profile + one other US Core profile.

**Submitter:**
- [x] This pull request describes why these changes were made
- [x] Internal ticket for this PR: https://oncprojectracking.healthit.gov/support/browse/FI-385
- [x] Internal ticket links to this PR
- [x] Internal ticket is properly labeled (Community/Program)
- [x] Internal ticket has a justification for its Community/Program label
- [x] Code diff has been reviewed for extraneous/missing code
- [x] Tests are included and test edge cases
- [x] Tests/code quality metrics have been run locally and pass

**Reviewer 1:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

**Reviewer 2:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

**Comment by arscan** [comment]
Is there a requirement somewhere that the server SHALL list all supported profiles in their capability statement if they support it?

**Comment by arscan** [comment]
I asked the question here: https://chat.fhir.org/#narrow/stream/179175-argonaut/topic/How.20do.20servers.20'claim.20conformance'.20to.20a.20profile.3F

The closest we could find right now is that the core specification says that a server SHOULD fill out that information, but it is not a requirement.  If we are given no more direction, we should either turn this into an optional test, or spit out a warning, and reference that should statement in the core spec: 

> Servers SHALL specify what resource types and operations are supported, and SHOULD also specify profiles for each resource type.  
> https://www.hl7.org/fhir/capabilitystatement.html#notes

**Comment by arscan** [comment]
This isn't backed as a requirement in US Core. What we could do is just check for RESOURCE level support here instead, as that is required.

**Comment by arscan** [comment]
And throw a warnings if they don't explicitely state profile support within the resource, as that is a SHOULD.

File: lib/app/modules/us_core_guidance/us_core_r4_capability_statement_sequence.rb [hunk 1]
```diff
@@ -55,9 +55,43 @@ class UsCoreR4CapabilityStatementSequence < CapabilityStatementSequence
         * [DSTU2 Conformance Statement](https://www.hl7.org/fhir/DSTU2/conformance.html)
       )
 
-      test 'FHIR server capability states JSON support' do
+      PROFILES = {
```
Comment (arscan) [inline lib/app/modules/us_core_guidance/us_core_r4_capability_statement_sequence.rb]: This is a little bit redundant with the data provided in app/helpers/validation.rb, and perhaps in other places.  I'm a little worried if these change we won't remember to update all the different places that these get referenced.  But this is a bigger problem then just this PR, so I'm good with leaving this as is.

Comment (Jammjammjamm) [reply lib/app/modules/us_core_guidance/us_core_r4_capability_statement_sequence.rb]: Yeah, we might want to come up with some general module-level structure for storing these kinds of module-wide knowledge/behaviors.
